### PR TITLE
Upgrade bg actions package and apply recommendations

### PR DESCRIPTION
**Pull request by nop33**
This is a problem we get often on Sentry. We also got several crash reports on Google in the past.

It doesn't seem to improve the dApp experience. I don't know if it makes it worse either. I also don't know if it fixes the error reports.

See:
- #914

### CODE REVIEW

File: apps/mobile-wallet/app.config.js [hunk 1]
```diff
@@ -37,7 +37,7 @@ export default {
       supportsTablet: true,
       bundleIdentifier: 'org.alephium.mobilewallet',
       infoPlist: {
-        BGTaskSchedulerPermittedIdentifiers: ['$(PRODUCT_BUNDLE_PACKAGE_TYPE)'],
+        BGTaskSchedulerPermittedIdentifiers: ['$(PRODUCT_BUNDLE_IDENTIFIER)'],
```
Comment (nop33) [inline apps/mobile-wallet/app.config.js:40]: This is mentioned in https://github.com/Rapsssito/react-native-background-actions/blob/master/INSTALL.md

image

File: patches/react-native-background-actions@4.0.1.patch [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+diff --git a/android/src/main/AndroidManifest.xml b/android/src/main/AndroidManifest.xml
+index b67ef4d1837363489a1749acb852997a6c93c1b5..147f04a080e4faf47069c7628bb541aa439f644d 100644
+--- a/android/src/main/AndroidManifest.xml
++++ b/android/src/main/AndroidManifest.xml
+@@ -1,6 +1,6 @@
+ <manifest xmlns:android="http://schemas.android.com/apk/res/android" package="com.asterinet.react.bgactions">
+     <uses-permission android:name="android.permission.FOREGROUND_SERVICE" />
+     <application>
+-        <service android:name=".RNBackgroundActionsTask"/>
++        <service android:name=".RNBackgroundActionsTask" android:foregroundServiceType="dataSync" android:stopWithTask="true" />
```
Comment (nop33) [inline patches/react-native-background-actions@4.0.1.patch:13]: Applying the same patch as the one we had before

File: apps/mobile-wallet/app.config.js (hunk 1 above)
Comment (mvaivre) [reply apps/mobile-wallet/app.config.js:40]: Ok, interesting

**Review by mvaivre — APPROVED** [review]
Good one!

**Pull request merged**